signer/local_cosigner.go

```diff
@@ -109,30 +100,34 @@ func (cosigner *LocalCosigner) SaveLastSignedState(signState SignStateConsensus)
 	return cosigner.lastSignState.Save(signState, &cosigner.lastSignStateMutex, true)
 }
 
-func NewLocalCosigner(cfg LocalCosignerConfig) *LocalCosigner {
+func NewLocalCosigner(
+	config *RuntimeConfig,
+	cosignerKey CosignerKey,
+	signState *SignState,
+	rsaKey rsa.PrivateKey,
+	peers []CosignerPeer,
+	address string,
+	total uint8,
+	threshold uint8,
+) *LocalCosigner {
```

Comment:
Eight positional arguments is a bit much. I think the single `LocalCosignerConfig` was a bit better.

Reply:
Using a single struct as a param for the constructor of another struct seems like unnecessary overhead to me. The potential zero values in the config struct require extra validate methods, whereas explicit parameters requires setting them all.

These `...Config` types have caused confusion for external contributors, which is the main reason I wanted to remove them.

Any thoughts on improving readability if we keep the `...Config` types?